deploy/charts/cert-manager/templates/cainjector-rbac.yaml

```diff
@@ -67,11 +67,14 @@ metadata:
     helm.sh/chart: {{ include "cainjector.chart" . }}
 rules:
   # Used for leader election by the controller
-  # TODO: refine the permission to *just* the leader election configmap
+  # Done: refine the permission to *just* the leader election configmaps
   - apiGroups: [""]
     resources: ["configmaps"]
-    verbs: ["get", "create", "update", "patch"]
-
+    resourceNames: ["cert-manager-cainjector-leader-election", "cert-manager-cainjector-leader-election-core"]
```

Comment:
Hm, I didn't realise we had two different configmaps used for leader election within the cainjector... do you know what the `core` one is used for?

Reply:
it seems that in cmd/cainjector/start.go line 107 a function runCertificateBaseInjector is made this one uses cert-manager-cainjector-leader-election configmap

On line 131 there is a func runSecretBasedInjector
This one uses the cert-manager-cainjector-leader-election-core configmap

if I then look at pkg/controller/cainjector/setup.go the function on lines 105 and 123 describe 2 types of injection controllers registrations.
so within the cert-manager-cainjector controller, there are actually 2 injection controllers running,
One for Certificate Resources and one for Secret Resources(`core`)

Reply:
Thanks for the in-depth investigation! We should probably fix this up at some point to only require one.. it doesn't make much sense to me that we'd have two different leader elections going on...

Reply:
https://github.com/jetstack/cert-manager/issues/2982